TestProviders/ReadyProvider.psm1

```diff
@@ -28,6 +28,27 @@ using Module ".\TestProvider.psm1"
 
 Class ReadyProvider : TestProvider
 {
+	[object] GetIPAddressFromIfconfig([string] $ifConfigInfo) {
+		Write-LogDbg "GetIPAddress $ifConfigInfo"
+		[regex] $re = "(?:[0-9]{1,3}\.){3}[0-9]{1,3}"
+		[string] $matchedIp = $re.Match($ifConfigInfo)
+		return $matchedIp
+	}
+
+	[void] SetInternalIPAddress([object] $AllVMData) {
+		$count = 0
+		foreach ($vmData in $AllVMData) {
+			$ifconfigInfo = Run-LinuxCmd -username $global:user -password $global:password -ip $($vmData.PublicIp) -port $($vmData.SSHPort) -command "ifconfig"
```

Comment:
1) We used 'ip addr show' as ifconfig will be depreciated.
2) This should be with runAsSudo